integration/caching_test.go

```diff
@@ -58,60 +58,64 @@ func testCaching(t *testing.T, context spec.G, it spec.S) {
 		context("online", func() {
 			it("reuses the node_modules layer", func() {
 				var err error
+				var container occam.Container
+
 				source, err = occam.Source(filepath.Join("testdata", "simple_app"))
 				Expect(err).NotTo(HaveOccurred())
 
-				build := pack.WithNoColor().Build.WithBuildpacks(nodeURI, yarnURI)
+				build := pack.WithNoColor().Build.WithBuildpacks(nodeURI, yarnURI, buildpackURI, buildPlanURI)
 
 				firstImage, firstLogs, err := build.Execute(name, source)
 				Expect(err).NotTo(HaveOccurred(), firstLogs.String)
 
 				imageIDs[firstImage.ID] = struct{}{}
 
-				Expect(firstImage.Buildpacks).To(HaveLen(2))
-				Expect(firstImage.Buildpacks[1].Key).To(Equal(buildpackInfo.Buildpack.ID))
-				Expect(firstImage.Buildpacks[1].Layers).To(HaveKey("yarn"))
-				Expect(firstImage.Buildpacks[1].Layers).To(HaveKey("modules"))
+				Expect(firstImage.Buildpacks).To(HaveLen(4))
+				Expect(firstImage.Buildpacks[2].Key).To(Equal(buildpackInfo.Buildpack.ID))
+				Expect(firstImage.Buildpacks[2].Layers).To(HaveKey("modules"))
 
-				container, err := docker.Container.Run.Execute(firstImage.ID)
+				container, err = docker.Container.Run.
+					WithCommand("ls -alR /layers/paketo-buildpacks_yarn-install/modules/node_modules").
+					Execute(firstImage.ID)
 				Expect(err).NotTo(HaveOccurred())
 
 				containerIDs[container.ID] = struct{}{}
 
-				Eventually(container).Should(BeAvailable())
+				Eventually(func() string {
+					cLogs, err := docker.Container.Logs.Execute(container.ID)
+					Expect(err).NotTo(HaveOccurred())
+					return cLogs.String()
+				}).Should(ContainSubstring("leftpad"))
 
 				secondImage, secondLogs, err := build.Execute(name, source)
 				Expect(err).NotTo(HaveOccurred(), secondLogs.String)
 
 				imageIDs[secondImage.ID] = struct{}{}
 
-				Expect(secondImage.Buildpacks).To(HaveLen(2))
-				Expect(secondImage.Buildpacks[1].Key).To(Equal(buildpackInfo.Buildpack.ID))
-				Expect(secondImage.Buildpacks[1].Layers).To(HaveKey("yarn"))
-				Expect(secondImage.Buildpacks[1].Layers).To(HaveKey("modules"))
+				Expect(secondImage.Buildpacks).To(HaveLen(4))
+				Expect(secondImage.Buildpacks[2].Key).To(Equal(buildpackInfo.Buildpack.ID))
+				Expect(secondImage.Buildpacks[2].Layers).To(HaveKey("modules"))
 
-				container, err = docker.Container.Run.Execute(secondImage.ID)
+				container, err = docker.Container.Run.
+					WithCommand("ls -alR /layers/paketo-buildpacks_yarn-install/modules/node_modules").
```

Comment:
Can we avoid hardcoding paketo-buildpacks_yarn-install and instead derive the layer name from the buildpack.toml info?